build-info-extractor-npm/src/main/java/org/jfrog/build/extractor/npm/extractor/NpmCommand.java

```diff
@@ -0,0 +1,73 @@
+package org.jfrog.build.extractor.npm.extractor;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jfrog.build.api.util.Log;
+import org.jfrog.build.client.ArtifactoryVersion;
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryClientBuilderBase;
+import org.jfrog.build.extractor.clientConfiguration.client.ArtifactoryBaseClient;
+import org.jfrog.build.extractor.npm.NpmDriver;
+import org.jfrog.build.extractor.npm.types.NpmPackageInfo;
+import org.jfrog.build.util.VersionCompatibilityType;
+import org.jfrog.build.util.VersionException;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Base class for npm install and npm publish commands.
+ *
+ * Created by Yahav Itzhak on 25 Nov 2018.
+ */
+abstract class NpmCommand implements Serializable {
+    private static final long serialVersionUID = 1L;
+    private static final ArtifactoryVersion MIN_SUPPORTED_ARTIFACTORY_VERSION = new ArtifactoryVersion("5.5.2");
+    private static final ArtifactoryVersion MIN_SUPPORTED_NPM_VERSION = new ArtifactoryVersion("5.4.0");
+
+    NpmPackageInfo npmPackageInfo = new NpmPackageInfo();
+    ArtifactoryClientBuilderBase clientBuilder;
+    ArtifactoryBaseClient client;
+    NpmDriver npmDriver;
+    List<String> args;
+    Path workingDir;
+    String repo;
+    Log logger;
+    Path path;
+
```

Comment:
The above class members should be probably declared as protected or private.

Reply:
They can't be private since NpmInstall and NpmPublish use them.
The 2 subclasses located in the same package as NpmCommand. In Java `Package` < `Protected` so it will break the `principle of least privilege`.